Modules/Scripted/VPAWVisualize/VPAWVisualize.py

```diff
@@ -322,25 +336,135 @@ def onApplyButton(self):
+        basename_repr = repr(basename)
+        props = {"name": basename, "singleFile": True, "show": False}
+
+        node = self.createNode(filename, basename_repr, props)
+        if node is None:
+            return
+        node_item = shNode.GetItemByDataNode(node)
+        # The node item is assigned the subject item as its parent.
+        shNode.SetItemParent(node_item, subject_item)
+        return
+
+    def addOutputWidgets(self, list_of_files):
+        # The subject hierarchy node can contain subject (patient), study (optionally),
+        # and node items.  slicer.mrmlScene knows how to find the subject hierarchy
+        # node.
+        shNode = slicer.mrmlScene.GetSubjectHierarchyNode()
+        # A subject item is created with the subject hierarchy node as its parent.
+        subject_name = (
+            self.ui.PatientPrefix.text if self.ui.PatientPrefix.text != "" else "All"
+        )
+        subject_item = shNode.CreateSubjectItem(shNode.GetSceneItemID(), subject_name)
+
+        # slicer knows how to find the subject hierarchy tree view.
+        shTV = slicer.qMRMLSubjectHierarchyTreeView()
+        # Tell the subject hierarchy tree view about its enclosing scene.
+        shTV.setMRMLScene(slicer.mrmlScene)
+        # Tell the subject hierarchy tree view that its root item is the subject item.
+        shTV.setRootItem(subject_item)
+
+        for filename in list_of_files:
+            self.addOutputWidget(shNode, subject_item, filename)
+
+        # Recursively set visibility and expanded properties of each item
+        def recurseVisibility(item, visibility, expanded):
+            # Useful functions for traversing items
+            # shNode.GetSceneItemID()
+            # shNode.GetNumberOfItems()
+            # shNode.GetNumberOfItemChildren(parentItem)
+            # shNode.GetItemByPositionUnderParent(parentItem, childIndex)
+            # shNode.SetItemExpanded(shNode.GetSceneItemID(), True)
+            shNode.SetItemDisplayVisibility(item, visibility)
+            shNode.SetItemExpanded(item, expanded)
+            for child_index in range(shNode.GetNumberOfItemChildren(item)):
+                recurseVisibility(
+                    shNode.GetItemByPositionUnderParent(item, child_index),
+                    visibility,
+                    expanded,
                 )
 
+        recurseVisibility(subject_item, True, True)
+
+        # Resize columns of the SubjectHierarchyTreeView
+        shTV.header().resizeSections(shTV.header().ResizeToContents)
+        # Force re-displaying of the SubjectHierarchyTreeView
+        slicer.mrmlScene.StartState(slicer.vtkMRMLScene.ImportState)
+        slicer.mrmlScene.EndState(slicer.vtkMRMLScene.ImportState)
+        # Make sure at least one input image (if any) is being viewed
+        if self.show_nodes:
+            slicer.util.setSliceViewerLayers(foreground=self.show_nodes[0], fit=True)
+        self.show_nodes = list()
+        # Center the 3D view
```

Comment:
Better to separate out centering the 3D view into its own function (in the logic class not the widget class), and call it somewhere that makes sense. Maybe calling it here makes sense -- that depends partly on what this function `addOutputWidgets` is renamed to.

Reply:
Done.